controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_types.go

```diff
@@ -234,6 +239,20 @@ type RemediationStrategy struct {
 	MinHealthyPeriod *metav1.Duration `json:"minHealthyPeriod,omitempty"`
 }
 
+// MachineNamingStrategy allows changing the naming pattern used when creating Machines.
+// InfraMachines & KubeadmConfigs will use the same name as the corresponding Machines.
+type MachineNamingStrategy struct {
+	// Template defines the template to use for generating the names of the Machine objects.
+	// If not defined, it will fallback to `{{ .kubeadmcontrolplane.name }}-{{ .random }}`.
+	// If the templated string exceeds 63 characters, it will be trimmed to 58 characters and will
+	// get concatenated with a random suffix of length 5.
+	// The templating mechanism provides the following arguments:
+	// * `.kubeadmcontrolplane.name`: The name of the KubeadmControlPlane object.
+	// * `.random`: A random alphanumeric string, without vowels, of length 5.
```

Comment:
Why expose this rather than always appending `-<random 5 chars>` at the end of the string?

Reply:
I see the following reasons:
* We already have namingStrategies in the ClusterClass that work the same way as the one introduced here (so consistency would be nice)
  * In that case we use the namingStrategies to calculate control plane / MD and MP names. In these cases it was desirable to not always end up with a random suffix. E.g. folks wanted to be able to have KCP names that are identical to Cluster names
* Another case would be if folks have the (admittedly strange) use case that the random part shouldn't be at the end of the name (there was one (!) person asking for it in Slack, they wanted a random prefix for some reason, xref: https://kubernetes.slack.com/archives/C8TSNPY4T/p1724389852680579)